src/Features/CSharp/Portable/RemoveUnnecessaryParentheses/CSharpRemoveUnnecessaryParenthesesDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.RemoveUnnecessaryParentheses;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.RemoveUnnecessaryParentheses
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class CSharpRemoveUnnecessaryParenthesesDiagnosticAnalyzer
+        : AbstractRemoveUnnecessaryParenthesesDiagnosticAnalyzer<SyntaxKind, ParenthesizedExpressionSyntax>
+    {
+        protected override SyntaxKind GetSyntaxNodeKind()
+            => SyntaxKind.ParenthesizedExpression;
+
+        protected override bool CanRemoveParentheses(
+            ParenthesizedExpressionSyntax parenthesizedExpression, SemanticModel semanticModel,
+            out PrecedenceKind precedenceKind, out bool clarifiesPrecedence)
+        {
+            return CanRemoveParenthesesHelper(parenthesizedExpression, semanticModel, out precedenceKind, out clarifiesPrecedence);
+        }
+
+        public static bool CanRemoveParenthesesHelper(
+            ParenthesizedExpressionSyntax parenthesizedExpression, SemanticModel semanticModel,
+            out PrecedenceKind precedenceKind, out bool clarifiesPrecedence)
+        {
+            var result = parenthesizedExpression.CanRemoveParentheses(semanticModel);
+            if (!result)
+            {
+                precedenceKind = default;
+                clarifiesPrecedence = false;
+                return false;
+            }
+
+            var inner = parenthesizedExpression.Expression;
+            var innerKind = inner.Kind();
+            var innerPrecedence = inner.GetOperatorPrecedence();
+            var innerIsSimple = innerPrecedence == OperatorPrecedence.Primary ||
+                                innerPrecedence == OperatorPrecedence.None;
+
+            ExpressionSyntax parentExpression;
+
+            switch (parenthesizedExpression.Parent)
+            {
+            case ConditionalExpressionSyntax _:
```

Comment:
I think our `case`s are usually indented

Reply:
ugh.  i hate this profile bug :)   I'll switch over (no pun intended).

Reply:
fixed.